src/runtime/test/json-id-to-manifest-test.ts

```diff
@@ -0,0 +1,120 @@
+/** @see LICENSE */
+
+import {assert} from '../../platform/chai-web.js';
+import {JsonldToManifest, supportedTypes} from '../converters/jsonldToManifest.js';
+import {fetch} from '../../platform/fetch-web.js';
+import {Predicate} from '../hot.js';
+import {Manifest} from '../manifest.js';
+
+describe('JsonldToManifest', () => {
+
+  const isValidManifest: Predicate<string> = (manifestStr: string): boolean => {
+    try {
+      Manifest.parse(manifestStr);
+    } catch (error) {
+      return false;
+    }
+    return true;
+  };
+
+  const asyncProductStr: Promise<string> = fetch('https://schema.org/Product.jsonld')
+    .then(r => r.text());
+
+  const asyncLocalBusinessStr: Promise<string> = fetch('https://schema.org/LocalBusiness.jsonld')
+    .then(r => r.text());
+
+  const messyOmit = (obj: object, p: Predicate<[string, unknown]>): void => {
+    Object.entries(obj).forEach((kv) => {
+      const key = kv[0];
+
+      if (p(kv)) {
+        delete obj[key];
+      }
+
+      if (typeof obj[key] === 'object' && obj[key] !== null) {
+        messyOmit(obj[key], p);
+      }
+    });
+
+  };
+
+  const omitKey = (obj: object, key: string) => messyOmit(obj, (kv) => kv[0] === key);
+
+  describe('convert', () => {
+    it('works on objects without @graph', async () => {
+      const valids = await asyncProductStr
+        .then((s: string) => JSON.parse(s))
+        .then((j: JSON) => j['@graph']);
```

Comment:
You very rarely need to use .then() syntax with await. This can simply be:
`const valids = JSON.parse(await asyncProductStr)['@graph'];`

Reply:
cebd9a2